remove old tasks periodically

elfkuzco opened the pull request:
## Rationale
To reduce the size of the DB backups, it is imperative to delete old tasks from the DB since they are not expected to be of any value once the task completes.

This resolves #1125


## Changes
- add function to remove tasks periodically to `periodic-tasks.py`
- add env vars to control removal.
     - `SHOULD_REMOVE_OLD_TASKS` determines whether tasks should be removed
     - `TASKS_OLDER_THAN` controls the number of days in the past to start deleting from

benoit74 commented on dispatcher/backend/src/periodic-tasks.py:
```diff
@@ -148,6 +155,49 @@ def staled_statuses(session: so.Session):
+        logger.info(f":: skipping removal of tasks older than {TASKS_OLDER_THAN} days")
+        return
+
+    logger.info(f":: removing tasks older than {TASKS_OLDER_THAN} days")
```
We should compute the "cutover" date and log it in this message, so that it is clear which tasks we've deleted.

elfkuzco commented:
Modified the `most_recent_task_id` constraint on the schedule to be set to null when a task is deleted and simplified the query for deletion. This would require an alembic migration to be made
